### PR TITLE
storage: system tests: manage created objects

### PR DESCRIPTION
**Pull request by stephenplusplus**
The main change here is forcing all created files and buckets to share a prefix. That way, they can be deleted all at once after all tests have completed.

### CODE REVIEW

File: system-test/storage.js [hunk 1]
```diff
@@ -28,63 +26,57 @@ var through = require('through2');
...
+    gzip: {
+      path: 'system-test/data/long-html-file.html.gz'
+    }
+  };
 
   before(function(done) {
     bucket.create(done);
   });
 
   after(function(done) {
```
Comment (callmehiphop) [inline system-test/storage.js]: Is this working for you locally? Mine keeps failing saying the bucket I'm trying to delete is not empty. Logic looks sound though, so maybe this is an upstream error?

Comment (stephenplusplus) [inline system-test/storage.js]: I had to run this in the terminal to clean up my old buckets. For some reason, they were un-deletable, unless I went through gsutil:

```sh
$ gsutil -m rm -r gs://gcloud-test-bucket-*
```

Comment (callmehiphop) [inline system-test/storage.js]: This worked for me, what happens if Travis runs into a similar issue?

Comment (stephenplusplus) [inline system-test/storage.js]: We'll have to have @jgeewax run it for us.

Comment (stephenplusplus) [inline system-test/storage.js]: I really can't speak to what the cause of this issue is (see https://github.com/GoogleCloudPlatform/gcloud-node/issues/968#issuecomment-159722959).

Comment (callmehiphop) [inline system-test/storage.js]: @jgeewax we've run into an issue with Travis, any chance you could fix that for us? Or just add us to the project so we don't have to ask you to fix it for us :stuck_out_tongue_winking_eye:

Comment (jgeewax) [inline system-test/storage.js]: Sent - sorry about the delay.

**Pull request merged**